include/deal.II/base/linear_index_iterator.h

```diff
@@ -0,0 +1,451 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_linear_index_iterator_h
+#define dealii_linear_index_iterator_h
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+/**
+ * Many classes in deal.II, such as FullMatrix, TransposeTable, and
+ * SparseMatrix, are fundamentally views into contiguous buffers: put another
+ * way, one can iterate across each of these classes with a random access
+ * iterator. This class is an iterator template, which can be used with any
+ * class that uses an array for its underlying storage. This class is
+ * essentially a simplified version of <code>boost::iterator_facade</code>
+ * that assumes <code>Accessor</code> provides certain features. The main
+ * intent of this class is that other classes can specify their own accessors
+ * and then use this class by inheriting with the curiously recurring template
+ * parameter (CRTP) technique, e.g.,
```

Comment:
curiously recurring template pattern (CRTP).

The phrasing ("inhereting with the [...] technique" sounds a bit odd to me, but you are certainly more the expert than me.
I would have shortened this to "and then use the curiously recurring template pattern (CRTP) technique".